src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -231,13 +446,25 @@ private Task ProcessHeadersFrameAsync<TContext>(IHttpApplication<TContext> appli
             }
 
             _receivedHeaders = true;
+            InputRemaining = HttpRequestHeaders.ContentLength;
 
-            Task.Run(() => base.ProcessRequestsAsync(application));
+            _appTask = Task.Run(() => base.ProcessRequestsAsync(application));
```

Comment:
Nit: It's probably better to make Http3Stream an IThreadPoolWorkItem like Http2Stream. You don't get a Task that way, but you can always set a TCS in the RequestProcessingEnded override.

Reply:
So Http3Stream is already an IThreadPoolWorkItem. There are two steps where you need to have an async step, one where you are on a Http3Connection and need to start a new stream (where the first IThreadPoolWorkItem), and next when you have received the headers and need to start the app func (This app task, which is calling HttpProtocol.ProcessRequestAsync).

What would you recommend here instead?

Reply:
The ThreadPool dispatching happens in a guaranteed order, so you can just check state in Http3Stream.Execute().